sourcegen-generator-kotlin/src/main/kotlin/io/micronaut/sourcegen/KotlinPoetSourceGenerator.kt

```diff
@@ -589,119 +606,497 @@ class KotlinPoetSourceGenerator : SourceGenerator {
+                    if (value is SwitchYieldCase) {
+                        builder.add("\n")
+                    } else {
+                        builder.add(";\n")
+                    }
+                }
+                if (elseExp != null) {
+                    builder.add("else -> ")
+                    builder.add(renderExpressionCode(objectDef, methodDef, elseExp))
+                }
+                builder.unindent()
+                builder.add("}")
+                return builder.build()
+            }
+            if (expressionDef is SwitchYieldCase) {
+                val builder: CodeBlock.Builder = CodeBlock.builder()
+                builder.add("{\n")
+                builder.indent()
+                val statement = expressionDef.statement
+                val flatten = statement.flatten()
+                check(!flatten.isEmpty()) { "SwitchYieldCase did not return any statements" }
+                val last = flatten[flatten.size - 1]
+                val rest: List<StatementDef> = flatten.subList(0, flatten.size - 1)
+                for (statementDef in rest) {
+                    builder.add(renderStatementCodeBlock(objectDef, methodDef, statementDef))
+                }
+                renderYield(builder, methodDef, last, objectDef)
+                builder.unindent()
+                builder.add("}")
+                val str: String = builder.build().toString()
+                // Render the body to prevent nested statements
+                return CodeBlock.of(str)
+            }
+            if (expressionDef is Condition) {
+                return CodeBlock.builder()
+                    .add(renderExpressionCode(objectDef, methodDef, expressionDef.left))
+                    .add(expressionDef.operator)
+                    .add(renderExpressionCode(objectDef, methodDef, expressionDef.right))
+                    .build()
+            }
+            if (expressionDef is PrimitiveInstance) {
+                return renderExpressionCode(objectDef, methodDef, expressionDef.value)
+            }
+            if (expressionDef is GetClassValue) {
+                val instanceExp = renderExpressionCode(objectDef, methodDef, expressionDef.instance)
+                return instanceExp.toBuilder().add(".javaClass").build()
+            }
+            if (expressionDef is HashCode) {
+                val instanceExp = renderExpressionCode(objectDef, methodDef, expressionDef.instance)
+                val type = expressionDef.instance.type()
+                if (type.isArray) {
+                    if (type is TypeDef.Array && type.dimensions > 1) {
+                        return instanceExp.toBuilder().add(".contentDeepHashCode()").build()
+                    }
+                    return instanceExp.toBuilder().add(".contentHashCode()").build()
+                }
+                return instanceExp.toBuilder().add(".hashCode()").build()
```

Comment:
`Objects.hashCode` implementation has a null case:
```
   public static int hashCode(Object o) {
        return o != null ? o.hashCode() : 0;
    }
```
but in this case we don't have it.

Reply:
What do you mean?

Reply:
I am not very familiar with kotlin sourcegen hrere, but for Java we have:
```java
           return renderExpression(objectDef, methodDef, instance.isNull().asConditionIfElse(
                ExpressionDef.constant(0),
                instance.invoke("hashCode", TypeDef.Primitive.BOOLEAN))
            );
```
Here we just call `o.hashCode()`?

Reply:
Yes, those methods are added externally, including a null check on `this`.